### ISSUE TITLE
NullProvider has been removed 

### ISSUE THREAD

**Issue by jorgheymans** [HubSpot/jackson-datatype-protobuf#15]
It seems that NullProvider was removed from jackson 2.8 onwards, this library is still using it. It's on their deprecated list for 2.7 (https://fasterxml.github.io/jackson-databind/javadoc/2.7/deprecated-list.html) but it doesn't say how to replace it.

**Comment by jhaber** [comment]
Thanks for the tip! I believe I found NullProvider by looking through the Jackson source and seeing how it handles null values, I'll take a look at the 2.8 source and see what it does now.

**Comment by cowtowncoder** [comment]
Worth noting that `NullProvider` has not been used sine 2.6, although class itself was left for bit longer.
I can probably help but I'd need to know what the intended usage was.

**Comment by jhaber** [comment]
I use it when I encounter a `null` value during deserialization of primitives, such as [here](https://github.com/HubSpot/jackson-datatype-protobuf/blob/4c478c21f0585093db3d378d05e61f0cdaa05ae7/src/main/java/com/hubspot/jackson/datatype/protobuf/ProtobufDeserializer.java#L197-L201). The intention was to avoid dealing directly with `DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES`; I thought going through `NullProvider` would be more robust than trying to duplicate the logic. In general I was trying to keep the protobuf deserializer behavior similar to `BeanDeserializer` (same exception with the same message, for example), but it seems like duplicating the code may be the way to go.

**Comment by cowtowncoder** [comment]
Ah. So as a convenient checker for whether `null`s may be coerced.
I did not realize anyone was using, outside of use by `SettableBeanProperty`, which is why it was removed.

Code duplication may indeed be the way to go, since the only (?) pluggable way I can think of is to access deserializers for relevant types (like `int` == `Integer.TYPE`), which is possible, but some work for maybe minor gain. Doing that would require implementing `ResolvableDeserializer` (method `resolve(...)` to fetch deserializers to delegate to (in theory it can be done from `deserialize(...)` but lookups are relatively expensive.
One benefit there would be to get fully standard checks and coercions (like unwrapping of single-element arrays).

Then again, handling of `null`s is the same for all primitives so maybe amount of duplication is limited.
